java/short-url/src/main/java/com/scdt/china/shorturl/service/impl/ShortUrlServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.scdt.china.shorturl.service.impl;
+
+import cn.hutool.core.util.StrUtil;
+import cn.hutool.crypto.SecureUtil;
+import com.scdt.china.shorturl.constant.CommonConstants;
+import com.scdt.china.shorturl.service.ShortUrlService;
+import com.scdt.china.shorturl.util.ShortUrlGenerator;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author：costa
+ * @date：Created in 2022/4/12 11:12
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ShortUrlServiceImpl implements ShortUrlService {
+    private final CacheManager cacheManager;
+
+    /**
+     * 获取短域名
+     */
+    public String getShortUrl(String url) {
+        String shortCode = ShortUrlGenerator.shortUrl(url);
+        Cache cache = cacheManager.getCache(CommonConstants.CODE2URL_MAP);
+
+        if (cache != null) {
+            String cacheUrl = cache.get(shortCode, String.class);
+            if (StrUtil.isEmpty(cacheUrl)) {
+                cache.put(shortCode, url);
+            } else if (url.equals(cacheUrl)) {
+                return shortCode;
+            } else {
+                log.info("第二次加密:" + shortCode );
+                shortCode = ShortUrlGenerator.shortUrl(SecureUtil.md5(url));
+                cacheUrl = cache.get(shortCode, String.class);
+                //理论上不会有url.equals(cacheUrl)的情况，不考虑，不为空就返回null
+                if (StrUtil.isEmpty(cacheUrl)) {
+                    cache.put(shortCode, url);
+                } else {//两次都有冲突返回NULL
+                    log.error("shortCode:" + shortCode );
+                    log.error("cacheUrl:" + cacheUrl );
+                    return null;
+                }
+            }
+
+        } else {//缓存为空则返回NULL
+            log.error("cache is null" );
+            return null;
+        }
+        return shortCode;
+    }
+
+    /**
+     * 获取长域名（从缓存中直接拿）
+     */
+    @Cacheable(value = CommonConstants.CODE2URL_MAP, key = "#shortCode")
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Generic Password' was found.
Severity: High

Would you like to exclude this secret from your status checks?
Tell us what to do with one of the following hashtags:

| Tag | Short Description |
| --- | --- |
| **#cycode_secret_ignore_here** | Applies to this request only |
| **#cycode_secret_ignore_everywhere** | Applies to this secret value for all repos in your organization |
| **#cycode_secret_revoked** | Applies to this secret value for all repos in your organization |
| **#cycode_secret_false_positive** | Applies to this secret value for all repos in your organization |

 <!-- 00393037-3561-4137-95e5-06f692c1e4b6 -->